src/Quizzer.API.App/Controllers/AnswerController.cs

```diff
@@ -4,75 +4,81 @@
+
             return Ok(answer);
         }
 
-        [HttpGet("byQuestion/{questionId:guid}", Name = "GetForQuestion")]
+        [HttpGet("ByQuestion/{questionId:guid}", Name = "[controller]/GetForQuestion")]
         public IEnumerable<AnswerDetailModel> GetForQuestion(Guid questionId) {
-            return _answerFacade.GetAnswersForQuestion(questionId);
+            return _facade.GetAnswersForQuestion(questionId);
         }
 
-        [HttpPut(Name = "UpdateAnswer")]
+        [HttpDelete("{id:guid}", Name = "[controller]/Delete")]
+        [ProducesResponseType(StatusCodes.Status404NotFound, Type = typeof(Guid))]
+        [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(Guid))]
+        public IActionResult Delete(Guid id) {
+            if (!_facade.Exists(id))
+                return NotFound(id);
+
+            _facade.Delete(id, DateTime.Now);
+            return Ok(id);
+        }
+
+        [HttpPut("{questionId:guid}", Name = "[controller]/Update")]
         [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(AnswerDetailModel))]
         [ProducesResponseType(StatusCodes.Status400BadRequest, Type = typeof(string))]
-        public IActionResult UpdateAnswer([FromBody] AnswerDetailModel answerModel, [FromBody] Guid questionId) {
+        public IActionResult Update([FromBody] AnswerDetailModel answerModel, Guid questionId) {
+            if (!_facade.Exists(questionId))
+                return BadRequest("Entity doesn't exist");
+
             Guid updatedId;
             try {
-                updatedId = _answerFacade.Save(answerModel, questionId, DateTime.Now);
+                updatedId = _facade.Save(answerModel, questionId, DateTime.Now);
             } catch (Exception ex) {
                 return BadRequest(ex.Message);
             }
             return Ok(updatedId);
         }
 
-        [HttpDelete("{id:guid}", Name = "DeleteAnswer")]
-        [ProducesResponseType(StatusCodes.Status404NotFound, Type = typeof(Guid))]
-        [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(Guid))]
-        public IActionResult DeleteAnswer(Guid id) {
-            AnswerDetailModel answer = _answerFacade.GetDetailById(id);
-            if (answer.Id != id)
-                return NotFound(id);
-
-            _answerFacade.Delete(id, DateTime.Now);
-            return Ok(id);
-        }
-
-        [HttpPost(Name = "CreateAnswer")]
+        [HttpPost("{questionId:guid}", Name = "[controller]/Create")]
         [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(Guid))]
         [ProducesResponseType(StatusCodes.Status400BadRequest, Type = typeof(string))]
-        public IActionResult CreateAnswer([FromBody] AnswerDetailModel answerModel, [FromBody] Guid questionId) {
-            Guid updatedId;
+        public IActionResult Create([FromBody] AnswerDetailModel answerModel, Guid questionId) {
+            if (_facade.Exists(questionId))
+                return BadRequest("Entity already exists");
```

Comment:
Same as above - i guess you wanted to pass id of answer.

Reply:
fixed